src/system/PermissionsModule/Api/AdminApi.php

```diff
@@ -1,550 +0,0 @@
-
-            $permissions = $query->getResult();
-
-            foreach ($permissions as $permission) {
-                $curseq = $permission['sequence'];
-
-                if ($curseq == $oldseq) {
-                    // we are dealing with the old value so make it the new value
-                    $curseq = $newseq;
-                } else {
-                    $curseq++;
-                }
-
-                $permission['sequence'] = (int)$curseq;
-            }
-        } else {
-            // The new sequence is lower in the list
-            // if the new requested sequence is bigger than
-            // the maximum sequence number then set it to
-            // the maximum number.  We don't want any spaces
-            // in the sequence.
-            if ($newseq > $maxseq) {
-                $newseq = (int)$maxseq;
-            }
-
-            $query = $this->entityManager->createQueryBuilder()
-                                         ->select('p')
-                                         ->from('ZikulaPermissionsModule:PermissionEntity', 'p')
-                                         ->where('p.sequence >= :oldseq')
-                                         ->andWhere('p.sequence <= :newseq')
-                                         ->setParameter('oldseq', $oldseq)
-                                         ->setParameter('newseq', $newseq)
-                                         ->orderBy('p.sequence', 'ASC')
-                                         ->getQuery();
-
-            $permissions = $query->getResult();
-
-            foreach ($permissions as $permission) {
-                $curseq = $permission['sequence'];
-
-                if ($curseq == $oldseq) {
-                    // we are dealing with the old value so make it the new value
-                    $curseq = $newseq;
-                } else {
-                    $curseq--;
-                }
-
-                $permission['sequence'] = (int)$curseq;
-            }
-        }
-
-        $this->entityManager->flush();
-
-        return true;
-    }
-
-    /**
-     * Get all security permissions schemas.
-     *
-     * @return array array if permission schema values
-     *
-     * @throws AccessDeniedException Thrown if the user doesn't have admin permissions over the module
-     */
-    public function getallschemas()
-    {
-        // Security check
-        if (!SecurityUtil::checkPermission('ZikulaPermissionsModule::', '::', ACCESS_ADMIN)) {
-            throw new AccessDeniedException();
-        }
-
-        BlockUtil::loadAll();
-        $schemas = SecurityUtil::getSchemas();
-        $modinfos = ModUtil::getAllMods();
-        foreach ($modinfos as $modinfo) {
-            if (!empty($modinfo['securityschema'])) {
-                $schemas = array_merge($schemas, $modinfo['securityschema']);
-            }
-        }
-        uksort($schemas, 'strnatcasecmp');
-        SecurityUtil::setSchemas($schemas);
-
-        return $schemas;
-    }
-}
```

Comment:
I suppose I have, but IMO it isn't useful to other modules so it isn't needed. All the functionality is replaced internally. just because a file (in Core-1.4.x) is _labelled_ as an API doesn't mean it is truly an *external* API. remember these files were seen as mainly collections of the DB interactions not as services to the outside world. Take a look at the provided (now deleted) methods. If you disagree, we can reinstate the file if you feel it is important.